modules/programs/starship.nix

```
@@ -87,25 +87,25 @@ in {
     };
 
     programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
-      if [[ $TERM != "dumb" && (-z $INSIDE_EMACS || $INSIDE_EMACS == "vterm") ]]; then
+      if [[ $TERM != "dumb" && (-z $INSIDE_EMACS || $INSIDE_EMACS == "vterm" || $STARSHIP_INSIDE_EMACS == "yes") ]]; then
```

I'd prefer not to add Ad hoc environment variables.
I think the following would work, tested on local an remote machine, with `shell-mode`, `ansi-term`, and `vterm`
```suggestion
      if [[ $TERM != "dumb" && "''${INSIDE_EMACS/*tramp*/tramp}" != "tramp" ]]; then
```

---

Tramp SETS TERM != dumb BY DEFAULT!!!!!!!!!!!

look this var definition up:
```
;;;###tramp-autoload
(defcustom tramp-terminal-type "dumb"
  "Value of TERM environment variable for logging in to remote host.
Because Tramp wants to parse the output of the remote shell, it is easily
confused by ANSI color escape sequences and suchlike.  Often, shell init
files conditionalize this setup based on the TERM environment variable."
  :group 'tramp
  :type 'string)
```

If the whole point of this is just to deal with tramp, we don't need to check for INSIDE_EMACS.

Also, generally speaking, having something that specifically sets thing for specific editors is just a bad approach. checking for TERM being dumb makes sense. Looking at INSIDE_EMACS does not.

The only way I did it the way I did was to try to make things as unintrusive as possible.

@marsam if you really want to do things the right way, then just remove the stupid check for INSIDE_EMACS

---

As I mentioned in the other issue, I added  the initial `INSIDE_EMACS` check to avoid Starship conflicting with Tramp. I used nixpkgs and the Tramp documentation as reference. 
Maybe it was not a correct/complete solution, but the beauty of Open Source/Free Software is that we together could figure out a better solution.

I guess checking TERM=dumb is enough, so I'd promote #4135 instead of this one.

---
Honestly, I think the way you reported the issue/fix is unnecessarily entitled and rude. Remember that home-manager maintainers are volunteers.

But that's it from me, I'm giving up home-manager. I'm silencing this repository, and I don't plan to contribute anymore.

---

> But that's it from me, I'm giving up home-manager. I'm silencing this repository, and I don't plan to contribute anymore.

seems a bit dramatic to me. I never attacked you personally in any way, to be clear, so I'm not sure why you feel so offended that I attacked an approach that you even admitted was simply from nixpkgs and have no personal interest in defending.

---

> Honestly, I think the way you reported the issue/fix is unnecessarily entitled and rude. Remember that home-manager maintainers are volunteers.

... and I'm one such volunteer. You should try not to get attached to code that you wrote/approaches you took, and be open to better solutions when they are available. It needs to be possible to criticize bad approaches so that we can identify when we have chosen one and work towards better alternatives.

---

@IvanMalison, while suggesting better approaches is always welcome, there's a way to do so. I am probably not one to give lessons in tact, but words like "I'm not sure why someone thought it was a good idea", "stupid check" or "Tramp SETS TERM != dumb BY DEFAULT!!!!!!!!!!!" are not an appropriate way to humbly suggest improvements, and can **reasonably** be construed as personal attacks and entitlement.

I'd appreciate it if you could tone it down a bit for your future contributions. [Not everyone has the same sensibilities.](https://www.mit.edu/~jcb/tact.html)

@marsam, I'm sorry to see you go; your contributions will always be welcome if you change your mind.

---

When I wrote:

> Tramp SETS TERM = dumb BY DEFAULT!!!!!!!!!!!

It was a reaction to the fact that there was further criticism of this approach by @marsam along with additional suggestions that we add some further explicit check for tramp related issues, when it had already been explained multiple times, including when the pull request was originally authored:

* https://github.com/nix-community/home-manager/pull/4116#issuecomment-1597599315
* https://github.com/nix-community/home-manager/pull/853#discussion_r336019047
 
that this check was not necessary.

Is adding a bunch of exclamation points really that inflammatory a response to that? I'm just making sure that attention is paid to the point that is now being made for the third time.

> I'm not sure why someone thought it was a good idea

I feel like you have to try to interpret this uncharitably to really find it offensive. It certainly CAN be interpreted in a sort of matter of fact way. I literally am not sure why someone thought it was a good idea and perhaps there REALLY WAS a good reason for this and I'm simply unaware of it, which is why I originally decided to leave the check as it was and simply allow an opt out.

> personal attacks

Personal attacks? Entitlement? It has to be okay to criticize ideas, and criticizing ideas is not the same thing as attacking the people that came up with them. 

> entitlement.

How am I sounding entitled?

I LITERALLY went out of my way to leave in place the existing bad behavior and simply allow users who, like me, want to opt out of what I regard as silly behavior.

If I were really entitled, I would have just proposed https://github.com/nix-community/home-manager/pull/4135 first and not just said "original behavior stupid and wrong". I was trying to be flexible and respectful to the originally existing behavior. That simply does not strike me as entitled at all.

ALSO, I LITERALLY have this fixed in my own config already. Kind of weird to call someone attempting to contribute a pull request entitled just because they are advocating for their view (which, lets be honest, seems to be ultimately correct).

---

To be fair, I'll grant that "stupid check" wasn't the nicest way to put it, and I regret saying that.

That said, @ncfavier I feel like overall, you came down FAR too hard on me in your comment, and your view that my tone was overly aggressive seems a bit over sensitive and not entirely fair to me.

---

@marsam I'm sorry to see you go too, I apologize for us not being able to cater for anyone in the project.

@IvanMalison I think you should disengage honestly, @ncfavier only tried to explain the context — please take into account the fact that someone felt offended by this, no need to take it further on someone else.